src/lib/checksum.c

```diff
@@ -150,3 +150,42 @@ uint32_t pseudo_header_initial(const int8_t *buf, size_t len)
   }
   return 0xFFFF0001;
 }
+
+uint8_t startoffset[2];
+uint8_t *prepare_packet(int8_t *buf, size_t len)
```

Comment:
Don't we have the protocol library to deal with parsing packets and pushing headers onto payload.
Honestly I like this hackish way better, but still it will be probably harder to maintain in the future.

Actually, could it be that the checksum calculation belongs to the protocol library?